src/v3/plugins/github/address.js

```diff
@@ -0,0 +1,196 @@
+// @flow
+
+import {NodeAddress, type NodeAddressT} from "../../core/graph";
+
+export opaque type GithubAddressT: NodeAddressT = NodeAddressT;
+
+const GITHUB_PREFIX = NodeAddress.fromParts(["sourcecred", "github"]);
+function githubAddress(...parts: string[]): GithubAddressT {
+  return NodeAddress.append(GITHUB_PREFIX, ...parts);
+}
+
+export type RepoAddress = {|
+  +type: "REPO",
+  +owner: string,
+  +name: string,
+|};
+export type IssueAddress = {|
+  +type: "ISSUE",
+  +repo: RepoAddress,
+  +number: string,
+|};
+export type PullAddress = {|
+  +type: "PULL",
+  +repo: RepoAddress,
+  +number: string,
+|};
+export type ReviewAddress = {|
+  +type: "REVIEW",
+  +pull: PullAddress,
+  +fragment: string,
+|};
+export type CommentAddress = {|
+  +type: "COMMENT",
+  +parent: IssueAddress | PullAddress | ReviewAddress,
+  +fragment: string,
+|};
+export type UserlikeAddress = {|
+  +type: "USERLIKE",
+  +login: string,
+|};
+
+export type StructuredAddress =
+  | RepoAddress
+  | IssueAddress
+  | PullAddress
+  | ReviewAddress
+  | CommentAddress
+  | UserlikeAddress;
+
+export function structure(x: GithubAddressT): StructuredAddress {
+  if (!NodeAddress.hasPrefix(x, GITHUB_PREFIX)) {
+    throw new Error(`Bad address: ${NodeAddress.toString(x)}`);
```

Comment:
I would prefer that these errors be extracted:

<details>
<summary>Patch</summary>

```diff
diff --git a/src/v3/plugins/github/address.js b/src/v3/plugins/github/address.js
index 5086038..10a8ada 100644
--- a/src/v3/plugins/github/address.js
+++ b/src/v3/plugins/github/address.js
@@ -48,21 +48,24 @@ export type StructuredAddress =
   | UserlikeAddress;
 
 export function structure(x: GithubAddressT): StructuredAddress {
+  function fail() {
+    return new Error(`Bad address: ${NodeAddress.toString(x)}`);
+  }
   if (!NodeAddress.hasPrefix(x, GITHUB_PREFIX)) {
-    throw new Error(`Bad address: ${NodeAddress.toString(x)}`);
+    throw fail();
   }
   const [_unused_sc, _unused_gh, kind, ...rest] = NodeAddress.toParts(x);
   switch (kind) {
     case "repo": {
       if (rest.length !== 2) {
-        throw new Error(`Bad address: ${NodeAddress.toString(x)}`);
+        throw fail();
       }
       const [owner, name] = rest;
       return {type: "REPO", owner, name};
     }
     case "issue": {
       if (rest.length !== 3) {
-        throw new Error(`Bad address: ${NodeAddress.toString(x)}`);
+        throw fail();
       }
       const [owner, name, number] = rest;
       const repo = {type: "REPO", owner, name};
@@ -70,7 +73,7 @@ export function structure(x: GithubAddressT): StructuredAddress {
     }
     case "pull": {
       if (rest.length !== 3) {
-        throw new Error(`Bad address: ${NodeAddress.toString(x)}`);
+        throw fail();
       }
       const [owner, name, number] = rest;
       const repo = {type: "REPO", owner, name};
@@ -78,7 +81,7 @@ export function structure(x: GithubAddressT): StructuredAddress {
     }
     case "review": {
       if (rest.length !== 4) {
-        throw new Error(`Bad address: ${NodeAddress.toString(x)}`);
+        throw fail();
       }
       const [owner, name, pullNumber, fragment] = rest;
       const repo = {type: "REPO", owner, name};
@@ -87,13 +90,13 @@ export function structure(x: GithubAddressT): StructuredAddress {
     }
     case "comment": {
       if (rest.length < 1) {
-        throw new Error(`Bad address: ${NodeAddress.toString(x)}`);
+        throw fail();
       }
       const [subkind, ...subrest] = rest;
       switch (subkind) {
         case "issue": {
           if (subrest.length !== 4) {
-            throw new Error(`Bad address: ${NodeAddress.toString(x)}`);
+            throw fail();
           }
           const [owner, name, issueNumber, fragment] = subrest;
           const repo = {type: "REPO", owner, name};
@@ -102,7 +105,7 @@ export function structure(x: GithubAddressT): StructuredAddress {
         }
         case "pull": {
           if (subrest.length !== 4) {
-            throw new Error(`Bad address: ${NodeAddress.toString(x)}`);
+            throw fail();
           }
           const [owner, name, pullNumber, fragment] = subrest;
           const repo = {type: "REPO", owner, name};
@@ -111,7 +114,7 @@ export function structure(x: GithubAddressT): StructuredAddress {
         }
         case "review": {
           if (subrest.length !== 5) {
-            throw new Error(`Bad address: ${NodeAddress.toString(x)}`);
+            throw fail();
           }
           const [owner, name, pullNumber, reviewFragment, fragment] = subrest;
           const repo = {type: "REPO", owner, name};
@@ -120,18 +123,18 @@ export function structure(x: GithubAddressT): StructuredAddress {
           return {type: "COMMENT", parent: review, fragment};
         }
         default:
-          throw new Error(`Bad address: ${NodeAddress.toString(x)}`);
+          throw fail();
       }
     }
     case "userlike": {
       if (rest.length !== 1) {
-        throw new Error(`Bad address: ${NodeAddress.toString(x)}`);
+        throw fail();
       }
       const [login] = rest;
       return {type: "USERLIKE", login};
     }
     default:
-      throw new Error(`Bad address: ${NodeAddress.toString(x)}`);
+      throw fail();
   }
 }
 
```

</details>

Reply:
Done